File: app/src/main/java/wang/relish/whatsnew/n/MainActivity.java
package wang.relish.whatsnew.n;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import wang.relish.whatsnew.R;
import wang.relish.whatsnew.n.ActiveNotificationsSampleActivity;
import wang.relish.whatsnew.n.DirectBootSampleActivity;
import wang.relish.whatsnew.n.MessagingServiceSampleActivity;
import wang.relish.whatsnew.n.MultiWindowSampleActivity;
import wang.relish.whatsnew.n.QuickSettingSampleActivity;
import wang.relish.whatsnew.n.ScopedDirectoryAccessSampleActivity;
import wang.relish.whatsnew.n.ShortcutsSampleActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_n);
    }

    //1 多窗口Playground
    public void goMultiWindowPlaygroundSample(View view) {
        goActivity(MultiWindowSampleActivity.class);
    }

    //2 活动通知
    public void goActiveNotificationsSample(View view) {
        goActivity(ActiveNotificationsSampleActivity.class);
    }

    //3 消息传递服务
    public void goMessagingServiceSample(View view) {
        goActivity(MessagingServiceSampleActivity.class);
    }

    //4 直接启动
    public void goDirectBootSample(View view) {
        goActivity(DirectBootSampleActivity.class);
    }

    //5 作用域目录访问
    public void goScopedDirectoryAccessSample(View view) {
        goActivity(ScopedDirectoryAccessSampleActivity.class);
    }

    //6 快捷方式
    public void goShortcutsSample(View view) {
        goActivity(ShortcutsSampleActivity.class);
    }

    //7 快速设置
    public void goQuickSettingSample(View view) {
        goActivity(QuickSettingSampleActivity.class);
    }

    // 作者GitHub地址
    public void goGitHub(View view) {
        goActivity(Intent.ACTION_VIEW, "https://github.com/relish-wang/DemoForN");
    }

    protected void goActivity(Class<?> clazz) {
        Intent intent = new Intent(this, clazz);
        startActivity(intent);
    }

    protected void goActivity(String action,String url){
        Intent intent = new Intent(action, Uri.parse(url));
        startActivity(intent);
    }
}

File: app/src/main/java/wang/relish/whatsnew/MainActivity.kt
package wang.relish.whatsnew

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import kotlinx.android.synthetic.main.activity_main_main.*
import wang.relish.whatsnew.n.MainActivity
import wang.relish.whatsnew.q.CameraActivity
import wang.relish.whatsnew.q.StorageActivity

/**
 * @author wangxin
 * @since 20191111
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_main)

        rvSamples.adapter = SampleAdapter(CARDS)
        rvSamples.layoutManager = GridLayoutManager(this, 3)
    }


    companion object {
        val CARDS = listOf(
                Card("n", MainActivity::class.java),
                Card("storage", StorageActivity::class.java),
                Card("camera", CameraActivity::class.java)
        )
    }

    data class Card(
            val name: String,
            val aty: Class<*>
    )

    class SampleAdapter(
            private val mData: List<Card>
    ) : RecyclerView.Adapter<ViewHolder>() {
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            val v = LayoutInflater.from(parent.context).inflate(
                    android.R.layout.activity_list_item, parent, false)
            return object : ViewHolder(v) {}
        }

        override fun getItemCount(): Int {
            return mData.size
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            val icon = holder.itemView.findViewById(android.R.id.icon) as View
            icon.visibility = View.GONE
            val textView = holder.itemView.findViewById(android.R.id.text1) as TextView
            textView.text = mData[position].name
            textView.background = ContextCompat.getDrawable(textView.context, R.drawable.btn_bg)
            holder.itemView.setOnClickListener {
                val intent = Intent(it.context, mData[position].aty)
                it.context.startActivity(intent)
            }
        }

    }
}
